File: 24-swap-nodes-in-pairs/swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return None

        dummy = ListNode()
        cur = dummy
        cur.next = head

        while cur.next and cur.next.next:
            first = cur.next
            second = cur.next.next
            
            cur.next = second
            first.next = second.next
            second.next = first
            cur = first

        return dummy.next

File: 24-swap-nodes-in-pairs/swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy = new ListNode(0, head);
        ListNode cur = dummy;

        while (cur.next != null && cur.next.next != null) {
            ListNode first = cur.next;
            ListNode second = cur.next.next;

            ListNode nxt = second.next;
            second.next = first;
            first.next = nxt;
            cur.next = second;

            cur = first;
        }

        return dummy.next;
    }
}